From fbef13f23e43216da7ece9c090153ff80adf7525 Mon Sep 17 00:00:00 2001
From: Denys Fedoryshchenko <denys.f@collabora.com>
Date: Wed, 11 Sep 2024 01:39:42 +0300
Subject: [PATCH] fix(musl/l2tp_session_free): Fix, likely typo

It is logical, that function should remain same,
not changed to free.

Signed-off-by: Denys Fedoryshchenko <denys.f@collabora.com>
---
 accel-pppd/ctrl/l2tp/l2tp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/accel-pppd/ctrl/l2tp/l2tp.c b/accel-pppd/ctrl/l2tp/l2tp.c
index 9fc2283c..bb1d1699 100644
--- a/accel-pppd/ctrl/l2tp/l2tp.c
+++ b/accel-pppd/ctrl/l2tp/l2tp.c
@@ -856,7 +856,7 @@ static void l2tp_tunnel_free_sessions(struct l2tp_conn_t *conn)
 #ifdef HAVE_FREE_FN_T
 	tdestroy(sessions, (__free_fn_t)l2tp_session_free);
 #else
-	tdestroy(sessions, free);
+	tdestroy(sessions, (void(*)(void *))l2tp_session_free);
 #endif
 	/* Let l2tp_session_free() handle the session counter and
 	 * the reference held by the tunnel.
